src/js/yeldaWebchatInjector.js

```diff
@@ -211,9 +217,72 @@ class YeldaChat {
       this.closeChat()
     } else if (event.data === 'openChat' || event.message === 'openChat') {
       this.openChat()
+    } else if (event.data && event.data.event && event.data.event === 'openLightGallery') {
+      if (!event.data.mediaSources || !event.data.mediaSources.length) {
+        return
+      }
+
+      this.handleLightGallery(event.data)
     }
   }
 
+  /**
+   * Create DynamicElements & Opens Light Gallery from the received mediaSources
+   * @param {Object} data - iframe data sent from webchat window
+   * @param {Array<String | Object>} data.mediaSources - Array of source for image or video
+   * @param {Object} data.labels - Labels for twitter & pinterest share text
+   */
+  handleLightGallery(data) {
+    // Get lightgallery container
+    let lightgalleryContainer = document.getElementById('lightgallery')
+
+    // If lightgalleryContainer not exits create it
+    if (!lightgalleryContainer) {
+      lightgalleryContainer = document.createElement('div')
+      lightgalleryContainer.id = 'lightgallery'
+      document.getElementById('yelda_iframe_container').appendChild(lightgalleryContainer)
+    }
+
+    // Create dynamic Elements from the mediaSources for the lightgallery
+    const dynamicElements = data.mediaSources.map(mediaSource => {
+      const shareURL = typeof mediaSource === 'string' ? mediaSource : mediaSource.urls[0]
+      let videoSources = ''
+
+      if (mediaSource.urls && mediaSource.urls.length) {
+        mediaSource.urls.forEach(url => {
+          videoSources += `<source src="${url}"></source>`
+        })
+      }
+
+      return {
+        tweetText: data.labels.twitterText,
+        pinterestText: data.labels.pinterestText,
+        facebookShareUrl: shareURL,
+        twitterShareUrl: shareURL,
+        pinterestShareUrl: shareURL,
+        googleplusShareUrl: shareURL,
+        ...(typeof mediaSource === 'string' && {
+          src: mediaSource,
+          href: mediaSource,
+        }),
+        ...(mediaSource.urls && videoSources && {
+          html: `<video class="lg-video-object lg-html5" controls="true" preload="none">${videoSources}</video>`,
+          href: mediaSource.urls[0]
+        }),
+        ...(mediaSource.cover && {
+          poster: mediaSource.cover
+        })
+      }
+    })
```

Comment:
1. Please follow existing convention, don't create not needed new names
```js
        tweetText: data.labels.twitterText, // => why new name ? 
        pinterestText: data.labels.pinterestText, // => :ok: 
```
```js
        tweetText: data.labels.tweetText,  // => :ok: 
        pinterestText: data.labels.pinterestText, // => :ok: 
```
2. Comment and keep it simple
```js
     return {
        ...data.labels, // { tweetText, pinterestText }
        facebookShareUrl: shareURL,
        // ...
}
```

But that's just for info, as thomas asked for no labels

3. use clear small functions with meaningfull names 
```js

/**
 * format object with social media shareUrl
 * @param {String} shareUrl url to share
 * @return {Object} shareUrl properties { facebookShareUrl, twitterShareUrl, pinterestShareUrl }
 */
getSharedUrlProperties(shareUrl) {
return {
        facebookShareUrl: shareURL,
        twitterShareUrl: shareURL,
        pinterestShareUrl: shareURL
 }
}

/**
 * Return object of image needed properties
 * @param {String} mediaSource image url
 * @return {Object} object of image needed properties {src, href, facebookShareUrl, twitterShareUrl, pinterestShareUrl }
 */
getImageProperties(mediaSource) {
return {
    src: mediaSource,
    href: mediaSource,
    ...this.getSharedUrlProperties(mediaSource)  // { facebookShareUrl, twitterShareUrl, pinterestShareUrl }
  }
}

/**
 * Return object of video needed properties
 * @param {Object} mediaSource video properties
 * @param {Array<String>} mediaSource.urls videos urls
 * @param {String} mediaSource.cover image cover, optional
 * @return {Object} object of image needed properties {html, href, facebookShareUrl, twitterShareUrl, pinterestShareUrl }
 */
getVideoProperties(mediaSource) {
const videoSources =  mediaSource.urls.reduce(acc, url) => {
    return `${acc}<source src="${url}"></source>`;
  }, '');

return {
     html: `<video class="lg-video-object lg-html5" controls="true" preload="none">${videoSources}</video>`,
     href: mediaSource.urls[0],
     ...(mediaSource.cover && {
      poster: mediaSource.cover // optional video cover preview
     }),
     ...this.getSharedUrlProperties(mediaSource.urls[0]) // { facebookShareUrl, twitterShareUrl, pinterestShareUrl }
  }
}
```

4. clearly manage different case and comment
```js
let mediaDetails = {}

// If mediaSource is a simple string, it means that the media is an image
if (typeof mediaSource === 'string') {
  mediaDetails = this.getImageProperties(mediaSource)
}
 // if urls array is defined, it means it's an array of videos with different video's formats to use as source
} else if(mediaSource.urls && mediaSource.urls.length) {
    mediaDetails = this.getVideoProperties(mediaSource)
}

return {
   ...mediaDetails // sharedURLs, href, html or source     
   }
```